### PR TITLE
Cleanup tests + fix accidental test overriding + remove AudioTestUtil + refactor TestUtil + improve TestUtil + improve TestBase

### PR DESCRIPTION
**Pull request by Taeir**
- Fixed tests that were accidentally overriding other tests by making test method names in abstract test classes more explicit.
- Changed some code to use mocks instead of anonymous classes
- Made TestUtil more capable, so it works correctly for the MainTest.
- Updated TestBase to recreate the Main mock for every test to avoid all problems.
- Removed AudioTestUtil as it is no longer used
- Updated all tests to not unnecessarily use the TestUtil to mock Main themselves, as the TestBase already does that.

### CODE REVIEW

File: src/test/java/nl/tudelft/contextproject/MainTest.java [hunk 1]
```diff
@@ -54,8 +53,8 @@ public void testMainDebugHud() {
 		Main mMock = mock(Main.class);
 		Main.setInstance(mMock);
 		String[] args = {"a", "--debugHud", "b"};
-	   Main.main(args);
-	   assertTrue(Main.isDebugHudShown());
+		Main.main(args);
```
Comment (Geweldig) [inline src/test/java/nl/tudelft/contextproject/MainTest.java:56]: Incorrect spacing, happens on other lines in this file aswel

**Comment by Geweldig** [comment]
Other than my one comment I have no complaints and think this should be merged.

File: src/test/java/nl/tudelft/contextproject/MainTest.java (hunk 1 above)
Comment (Taeir) [inline src/test/java/nl/tudelft/contextproject/MainTest.java:56]: I fixed the spacing, the old spacing was incorrect.

Comment (Geweldig) [inline src/test/java/nl/tudelft/contextproject/MainTest.java:56]: Huh you're right, just checked in my IDE. Wonder why GitHub shows it as broken.

**Comment by WytzeElhorst** [comment]
I agree, ready to merge imo

**Pull request merged**